option http '-e' and '-k' canot be used together

hello!   
I have found the issue shown below. 
  
This one works: 
`./sensu-base-checks http -e 5d -u https://www.google.com`

This one works: 
`./sensu-base-checks http -k -u https://www.google.com`

This one does **NOT** work:
`./sensu-base-checks http -e 5d -k -u https://www.google.com`

this is the error that it throws:

```
panic: runtime error: index out of range [0] with length 0

goroutine 19 [running]:
main.(*httpConfig).httpClient.func1(0xc000096480, 0x2, 0x2, 0x0, 0x0, 0x0, 0x2e1b213827b70a85, 0x53380d134d2c6dfc)
	/Users/js/src/github.com/julian7/sensu-base-checks/cmd/sensu-base-checks/http.go:212 +0xa4
crypto/tls.(*Conn).verifyServerCertificate(0xc000080380, 0xc000096480, 0x2, 0x2, 0x92b, 0x0)
	/usr/local/Cellar/go/1.14/libexec/src/crypto/tls/handshake_client.go:823 +0x4b3
crypto/tls.(*clientHandshakeStateTLS13).readServerCertificate(0xc000179e20, 0x0, 0x0)
	/usr/local/Cellar/go/1.14/libexec/src/crypto/tls/handshake_client_tls13.go:433 +0x15e
crypto/tls.(*clientHandshakeStateTLS13).handshake(0xc000179e20, 0xc0000c2000, 0x0)
	/usr/local/Cellar/go/1.14/libexec/src/crypto/tls/handshake_client_tls13.go:85 +0x1ee
crypto/tls.(*Conn).clientHandshake(0xc000080380, 0x0, 0x0)
	/usr/local/Cellar/go/1.14/libexec/src/crypto/tls/handshake_client.go:196 +0x4d9
crypto/tls.(*Conn).Handshake(0xc000080380, 0x0, 0x0)
	/usr/local/Cellar/go/1.14/libexec/src/crypto/tls/conn.go:1340 +0xcc
net/http.(*persistConn).addTLS.func2(0x0, 0xc000080380, 0x0, 0xc00009c1e0)
	/usr/local/Cellar/go/1.14/libexec/src/net/http/transport.go:1453 +0x42
created by net/http.(*persistConn).addTLS
	/usr/local/Cellar/go/1.14/libexec/src/net/http/transport.go:1449 +0x1aa
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 Fixed:
 
 * Try all available NTP servers until one answers. It can fix intermittent NTP issues querying only one server.
+* Deep certificate expiration check
+* Allow certificate expiration to be checked while skipping cert verification [#2]
 
 ## [v0.3.1] - Dec 23, 2020
 

--- a/cmd/sensu-base-checks/http.go
+++ b/cmd/sensu-base-checks/http.go
@@ -23,25 +23,25 @@ import (
 )
 
 type httpConfig struct {
-	URL        string
-	Timeout    string
-	timeout    time.Duration
-	Headers    []string
-	Insecure   bool
-	Certfile   string
-	CAfile     string
-	Expiry     string
-	expiry     time.Time
-	Method     string
-	Metrics    bool
-	Response   uint
-	Redirect   string
-	UserAgent  string
-	Data       string
-	JSONkey    string
-	JSONval    string
-	certExpiry time.Time
-	tracer     *measurements.HTTPTracer
+	URL       string
+	Timeout   string
+	timeout   time.Duration
+	Headers   []string
+	Insecure  bool
+	Certfile  string
+	CAfile    string
+	Expiry    string
+	expiry    time.Time
+	Method    string
+	Metrics   bool
+	Response  uint
+	Redirect  string
+	UserAgent string
+	Data      string
+	JSONkey   string
+	JSONval   string
+	certList  []*x509.Certificate
+	tracer    *measurements.HTTPTracer
 }
 
 func httpCmd() *cobra.Command {
@@ -172,11 +172,14 @@ func (conf *httpConfig) Run(cmd *cobra.Command, args []string) error {
 	}
 
 	if len(conf.Expiry) != 0 {
-		if conf.certExpiry.Before(conf.expiry) {
-			return sensulib.Warn(fmt.Errorf(
-				"certificate will expire in %s",
-				durafmt.Parse(time.Until(conf.certExpiry)).LimitFirstN(4).String(),
-			))
+		for _, cert := range conf.certList {
+			if cert.NotAfter.Before(conf.expiry) {
+				return sensulib.Warn(fmt.Errorf(
+					"certificate %s will expire in %s",
+					cert.Subject,
+					durafmt.Parse(time.Until(cert.NotAfter)).LimitFirstN(4).String(),
+				))
+			}
 		}
 	}
 
@@ -208,8 +211,13 @@ func (conf *httpConfig) httpClient() (*http.Client, error) {
 	}
 
 	if len(conf.Expiry) != 0 {
-		tlsconfig.VerifyPeerCertificate = func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
-			conf.certExpiry = verifiedChains[0][0].NotAfter
+		tlsconfig.VerifyConnection = func(cs tls.ConnectionState) error {
+			conf.certList = make([]*x509.Certificate, len(cs.PeerCertificates))
+
+			for idx, cert := range cs.PeerCertificates {
+				conf.certList[idx] = cert
+			}
+
 			return nil
 		}
 	}

